### ISSUE TITLE
Move wait logic to standalone command

### ISSUE THREAD

**Issue by seaneagan** [GoogleContainerTools/kpt#525]
Statusing of documents is not only useful at the time of application, but also at any point in the document's lifecycle, so it would be useful to extract the wait logic into its own command. Instead of:

```
kpt live apply --wait-for-reconcile  --wait-timeout=15m --wait-polling-period=5s
```
it would be:
```
kpt live apply DIR 
kpt live wait --timeout=15m --polling-period=5s DIR
...
# run again for debugging purposes etc.
kpt live wait --polling-period=5s DIR
```

When using DIR, it could use the same mechanism as pruning to identify the existing applied document set:

`cli-utils.sigs.k8s.io/inventory-id: test-group`

Related questions for the wait logic:

1. Has using resource watches instead of polling been considered?
2. Doesn't there need to be something to additionally ensure that the e.g. `status.observedGeneration` matches the `metadata.generation`? Relying on only conditions would mean that potentially the status condition won't have been updated yet since the document application.
2. What about allowing metadata on CRDs for when they can't break backward compatibility to match the standard status conditions:

```
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: mycr.example.org
  annotations:
    cli-utils.sigs.k8s.io/reconciling-condition: Succeeded!=true
    cli-utils.sigs.k8s.io/stalled-condition: Failed=true
```

**Comment by mortent** [comment]
We are planning on adding an independent command for printing status. In fact, if you look at the https://github.com/kubernetes-sigs/cli-utils repo where most of the code for apply, prune and status live, there is already a status command https://github.com/kubernetes-sigs/cli-utils/blob/master/cmd/status/status.go. It still needs some work to fit properly into kpt, and taking advantage of the prune inventory is one of those things. We are working on it.

We have thought about using watches instead of polling for status. The status readers have been designed to look similar to controllers partly to make this possible in the future. But it would make things more complicated and it is not clear that it would be a better solution. Using polling we are able to rely heavily on LIST calls on individual namespaces rather than many GET calls.

You are correct that `status.ObservedGeneration` is important to avoid race conditions. The documentation for the kstatus library in cli-utils does [mention this](https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/kstatus/README.md), as does the [kpt docs](https://googlecontainertools.github.io/kpt/reference/live/apply/#recommendations-for-crds).

Supporting configurable rules for computing status through metadata is something we want to support. It probably is something that will take some time as we have many features in our pipeline which we feel are more important.

**Comment by seaneagan** [comment]
Fixed via #952
